<!-- -*- mode: XML -*- -->
<launch>
<arg name="sonarAddress" value="127.0.0.1"/>
<arg name="sonarPort" value="8000"/>
<arg name="loggerPath" value="/opt/Poseidon/www/webroot/record/"/>
<arg name="configPath" value="/opt/Poseidon/config.txt"/>

<node pkg="i2c_controller" name="i2c_controller" type="i2c_controller_node" output="screen" respawn="true" respawn_delay="1"/>
<node pkg="power_management" name="power_management" type="power_management_node" output="screen" respawn="true" respawn_delay="1"/>
<node pkg="raspberrypi_vitals" name="Vitals" type="raspberrypi_vitals_node" output="screen" respawn="true" respawn_delay="1"/>

<node pkg="hydroball_config_websocket" name="hydroball_config_websocket" type="hydroball_config_websocket_node" args="$(arg configPath)" output="screen"/>

<node pkg="logger" name="logger" type="logger_text_node" output="screen" args="$(arg loggerPath)" respawn="true">
	<param name="fileExtensionForGpsDatagram" value=".ubx"/>
</node>

<node pkg="imu_dummy" name="IMU" type="imu_dummy_node" output="screen"/>

<node pkg="gnss_dummy" name="GNSS" type="gnss_dummy_node" output="screen" /> 

<!--<node pkg="sonar_nmea_0183_tcp_client" name="sonar" type="nmea_network_node" output="screen" args="$(arg sonarAddress) $(arg sonarPort)"/>-->
<node pkg="sonar_dummy" name="sonar" type="sonar_dummy_node" output="screen" />

<node pkg="state_controller" name="state_controller" type="state_controller_node" output="screen"/>

<node pkg="hydroball_data_websocket" name="hydroball_data_websocket" type="hydroball_data_websocket_node" args="$(arg loggerPath)" output="screen"/>

<node pkg="hydroball_files_websocket" name="hydroball_files_websocket" type="hydroball_files_websocket_node" args="$(arg loggerPath)" output="screen"/>

<!--<node pkg="wifi_config" name="wifiConfig" type="wifi_config_node" output="screen"/>-->

<node pkg="diagnostics" name="Diagnostics" type="diagnostics_node" output="screen"/>


</launch>
